lib/model/katex.dart

```diff
@@ -0,0 +1,576 @@
+      // in katex.scss, but we encounter them in the generated HTML.
+      switch (spanClass) {
+        case 'mord':
+        case 'mopen':
+          classFound = true;
+      }
+
+      if (!classFound) _logError('KaTeX: Unsupported CSS class: $spanClass');
+
+      index++;
+    }
+
+    String? text;
+    List<KatexNode>? spans;
+    if (element.nodes case [dom.Text(:final data)]) {
+      text = data;
+    } else {
+      spans = _parseChildSpans(element);
+    }
+    if (text == null && spans == null) throw KatexHtmlParseError();
+
+    final inlineStyles = _parseSpanInlineStyles(element);
+
+    return KatexNode(
+      styles: inlineStyles != null
+        ? styles.merge(inlineStyles)
+        : styles,
+      text: text,
+      nodes: spans);
+  }
+
+  KatexSpanStyles? _parseSpanInlineStyles(dom.Element element) {
+    if (element.attributes case {'style': final styleStr}) {
+      // `package:csslib` doesn't seem to have a way to parse inline styles:
+      //   https://github.com/dart-lang/tools/issues/1173
+      // So, workaround that by wrapping it in a universal declaration.
+      final stylesheet = css_parser.parse('*{$styleStr}');
+      if (stylesheet.topLevels case [css_visitor.RuleSet() && final rule]) {
+        double? heightEm;
+        double? verticalAlignEm;
+
+        for (final declaration in rule.declarationGroup.declarations) {
+          if (declaration case css_visitor.Declaration(
+            :final property,
+            expression: css_visitor.Expressions(
+              expressions: [css_visitor.Expression() && final expression]),
+          )) {
+            switch (property) {
+              case 'height':
+                heightEm = _getEm(expression);
+                if (heightEm != null) continue;
+
+              case 'vertical-align':
+                verticalAlignEm = _getEm(expression);
+                if (verticalAlignEm != null) continue;
+            }
+
+            // TODO handle more CSS properties
+            _logError('KaTeX: Unsupported CSS property: $property of '
+              'type ${expression.runtimeType}');
+          } else {
+            throw KatexHtmlParseError();
+          }
+        }
+
+        return KatexSpanStyles(
+          heightEm: heightEm,
+          verticalAlignEm: verticalAlignEm,
+        );
+      } else {
+        throw KatexHtmlParseError();
+      }
+    }
+    return null;
+  }
+
+  double? _getEm(css_visitor.Expression expression) {
+    if (expression is css_visitor.EmTerm && expression.value is num) {
+      return (expression.value as num).toDouble();
+    }
+    return null;
+  }
+}
+
+enum KatexSpanFontWeight {
+  bold,
+}
+
+enum KatexSpanFontStyle {
+  normal,
+  italic,
+}
+
+enum KatexSpanTextAlign {
+  left,
+  center,
+  right,
+}
+
+class KatexSpanStyles {
+  double? heightEm;
+  double? verticalAlignEm;
+
+  String? fontFamily;
+  double? fontSizeEm;
+  KatexSpanFontStyle? fontStyle;
+  KatexSpanFontWeight? fontWeight;
```

Comment:
nit: put these in same order as the enums are defined

(And I think weight, then style, is the preferred way around: one says "bold italic", e.g. in the names of the fonts this PR adds, not "italic bold")